File: NapakalakiRuby/lib/numeric_bad_consequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'bad_consequence'

module NapakalakiGame
  class NumericBadConsequence < BadConsequence
    
    MAXTREASURES = 10
      
    def initialize(t, l, nVisible, nHidden)
      bc = super(t, l)
      @nVisibleTreasures=nVisible
      @nHiddenTreasures=nHidden
    end
  
    def getNVisibleTreasures
      @nVisibleTreasures
    end
  
    def getNHiddenTreasures
      @nHiddenTreasures
    end
    
    def isEmpty
      vacio=false
      if @nVisibleTreasures==0 && @nHiddenTreasures==0
        vacio = true
      end
      vacio
    end
  
    def substractVisibleTreasure(t)
      if @nVisibleTreasures > 0 then
        @nVisibleTreasures -= 1
      end
    end 
    
    def substractHiddenTreasure(t)
      if @nHiddenTreasures > 0 then
        @nHiddenTreasures -= 1
      end
    end
    
    def adjustToFitTreasureLists(v, h)
      n_visible = @nVisibleTreasures
      n_hidden = @nHiddenTreasures
      if v.size < n_visible then
        n_visible = v.size
      end
      if h.size < n_hidden then
      n_hidden = h.size
      end
      bc = NumericBadConsequence.new(@text, @levels, n_visible, n_hidden)
      bc
    end

    def to_s
      cadena = super
      cadena += "\nRandom_Visible_Treasures_Down = #{@nVisibleTreasures} / Random_Hidden_Treasures_Down = #{@nHiddenTreasures}"
      cadena
    end
    
      public_class_method :new
  
  end
end

File: NapakalakiJava/src/NapakalakiGame/NumericBadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.ArrayList;

/**
 *
 * @author danibolanos & jomabose
 */
public class NumericBadConsequence extends BadConsequence{
        private int nVisibleTreasures;
        private int nHiddenTreasures;
    
        public NumericBadConsequence(String t, int l, int nVisible, int nHidden){
            super(t, l);
            nVisibleTreasures = nVisible;
            nHiddenTreasures = nHidden;
        }
        
        public int getNVisibleTreasures(){
            return nVisibleTreasures;
        }
    
        public int getNHiddenTreasures(){
            return nHiddenTreasures;
        }
        
       @Override
       public boolean isEmpty(){
            boolean vacio=false;
            
            if (getNVisibleTreasures()==0 && getNHiddenTreasures()==0)
                vacio=true;
            
            return vacio;
       }
       
       @Override
        public void substractVisibleTreasure(Treasure t){
            if(getNVisibleTreasures()>0)
                nVisibleTreasures--;
        }
        
        @Override
        public void substractHiddenTreasure(Treasure t){
            if(getNHiddenTreasures()>0)
                nHiddenTreasures--;
        }
        
        @Override
        public BadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v, ArrayList<Treasure> h){
            int nVisible = nVisibleTreasures;
            int nHidden = nHiddenTreasures;
            
            if(v.size() < nVisible)
                nVisible = v.size();
            
            if(h.size() < nHidden)
                nHidden = h.size();
            BadConsequence bc = new NumericBadConsequence(text, levels, nVisible, nHidden);
            
            return bc;
        }
        
        @Override
        public String toString(){
            String cadena = super.toString();
            cadena += "\nRandom_Visible_Treasures_Down = " + Integer.toString(nVisibleTreasures) 
            + " / Random_Hidden_Treasures_Down = " + Integer.toString(nHiddenTreasures);
                
            return cadena;
        }
}
